data-prepper-plugins/saas-source-plugins/source-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/source_crawler/base/LeaderProgressState.java

```diff
@@ -0,0 +1,19 @@
+package org.opensearch.dataprepper.plugins.source.source_crawler.base;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.opensearch.dataprepper.plugins.source.source_crawler.coordination.state.AtlassianLeaderProgressState;
+import org.opensearch.dataprepper.plugins.source.source_crawler.coordination.state.CrowdStrikeLeaderProgressState;
+
+import java.time.Instant;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = AtlassianLeaderProgressState.class),
+        @JsonSubTypes.Type(value = CrowdStrikeLeaderProgressState.class)
+})
+public interface LeaderProgressState {
```

Comment:
does it not store information about failures too?

plz add java docs on what is "pollTime"